docs/proposals/28-autonomous-shoot-clusters.md

```diff
@@ -0,0 +1,497 @@
+Hence, we propose to use a [KinD cluster](https://kind.sigs.k8s.io), deploy the needed components into it, and create the necessary `Infrastructure`, `MachineClass`, and `MachineDeployment` resources.
+This way, these components can be reused as-is, without big changes or re-wiring of these complex components to run in-process with a fake client (as we do during the `gardenadm init` flow for the simpler components).
+Once the controllers have reconciled successfully, we can export the created infrastructure state and the `Machine` resources, bootstrap the autonomous shoot cluster similar to the [general flow](#commands) also needed for the [High Touch scenario](#high-touch), and import the state again into the created cluster.
+
+> [!NOTE]
+> Since worker node management will be taken over by `machine-controller-manager`, [step 4](#commands) (joining worker nodes) from above list can be omitted.
+> `machine-controller-manager` will also take over the creation of the control plane nodes, however, bootstrapping and joining them must still be performed by the user (meaning that [step 3](#commands) remains).
+
+Since we assume that this scenario will be the prominent one, we plan to add another command, `gardenadm bootstrap`, which shall simplify these steps.
+
+##### `gardenadm bootstrap`
+
+This command creates the virtual network and machines to be used for the control plane of the autonomous shoot cluster.
+It uses an existing KinD cluster to temporarily deploy existing Gardener components (unaware of the autonomous shoot cluster use-case).
+These components, such as provider extensions and `machine-controller-manager`, are used to create the necessary infrastructure and machines.
+
+The high-level steps are as follows:
+
+1. Deploy `gardener-resource-manager` to be able to create provider extensions.
+2. Deploy the required provider extension.
+3. Apply the `Infrastructure` resource to create the virtual network.
+4. Deploy `machine-controller-manager` to be able to create the machines.
+5. Apply the `Worker` resource to create the control plane nodes.
+6. Wait for the machines to be created.
+7. Stop `machine-controller-manager` to prevent deletion of the machines as they will not become ready since there is no Kubernetes cluster yet to join.
+8. Create a `Bastion` resource to be able to access the machines and execute `gardenadm init` (can be later optimized via `cloud-init` through a cloud config that invokes `gardenadm init` directly when creating the machines).
+
+The states of the `Infrastructure` and `Machine` resources need to be exported from the KinD cluster.
+Once the autonomous shoot cluster control plane is running, the state should be imported so that it can be managed by the provider extension and `machine-controller-manager` respectively.
+
+The KinD cluster can be deleted afterwards.
+
+### Disaster Recovery
```

Comment:
This chapter applies to the `High touch` scenario only, doesn't it?
In a `Medium touch` scenario, I could imagine that there is a cluster-autoscaler and an MCM which could handle node-failures automatically. In this case, hard-disk failures and node-failures could be independent from each other, because the hard-disk for etcd could be a mounted block storage.

Depending on the setup, it could even resolve data center failures.

Reply:
cc @ScheererJ

Reply:
From my point of view the chapter applies to both `high touch` and `medium touch` scenarios. That being said the reasons for certain outages can be different. For example, while a power outage can cause node failures in `high touch` and `medium touch` scenarios a hypervisor issue can only affect the `medium touch` scenario.

The GEP currently does not envision the level of automation required for full node replacement. Therefore, even in the `medium touch` scenario automatic replacement of failed control plane nodes is not included as it is not sufficient to replace the node, but `gardenadm join` also needs to be executed with the right set of parameters. This would be part of a `low/no touch` scenario, which we touched on in https://github.com/gardener/gardener/blob/6e7ef0b8753b9abdc98b63cc12d867b11522a2ec/docs/proposals/28-autonomous-shoot-clusters.md#autonomous-shoot-clusters-for-end-users.

In summary, some issues can eventually be addressed in a lower touch scenario, but in the beginning `high touch` and `medium touch` are fairly similar in terms of disaster recovery as there is not much automation.